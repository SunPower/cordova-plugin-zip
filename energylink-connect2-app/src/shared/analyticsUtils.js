import {
  __,
  compose,
  curry,
  gt,
  isEmpty,
  join,
  lensIndex,
  map,
  not,
  over,
  props,
  propSatisfies,
  toLower,
  toUpper,
  when
} from 'ramda'

export const isNotEmpty = compose(not, isEmpty)

export const capitalizeWord = when(
  propSatisfies(gt(__, 0), 'length'),
  compose(join(''), over(lensIndex(0), toUpper), toLower)
)

const stringValueOrEmptyString = stringValue => stringValue || ''

const getStringProps = curry(compose(map(stringValueOrEmptyString), props))

export const getUserProfile = getStringProps([
  'uniqueId',
  'firstName',
  'lastName',
  'email',
  'dealerName',
  'recordType'
])
