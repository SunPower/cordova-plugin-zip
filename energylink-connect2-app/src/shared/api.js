import Swagger from 'swagger-client'
let apiParty, apiSite, apiFirmware, apiScanSN

const requestOptions = access_token => ({
  requestInterceptor: req => {
    if (access_token) req.headers['Authorization'] = `Bearer ${access_token}`
    return req
  }
})

export async function getApiFirmware() {
  if (!apiFirmware)
    apiFirmware = await Swagger(process.env.REACT_APP_SWAGGER_FIRMWARE)
  return apiFirmware
}
export async function getApiPVS() {
  return await Swagger(process.env.REACT_APP_PVS_ADDRESS)
}
export async function getApiDevice(access_token) {
  return await Swagger(
    process.env.REACT_APP_SWAGGER_DEVICE,
    requestOptions(access_token)
  )
}
export async function getApiParty(access_token) {
  apiParty = await Swagger(
    process.env.REACT_APP_SWAGGER_PARTY,
    requestOptions(access_token)
  )
  return apiParty
}
export async function getApiSearch(access_token) {
  return await Swagger(
    process.env.REACT_APP_SWAGGER_SEARCH,
    requestOptions(access_token)
  )
}
export async function getApiSite(access_token) {
  if (!apiSite)
    apiSite = await Swagger(
      process.env.REACT_APP_SWAGGER_SITE,
      requestOptions(access_token)
    )
  return apiSite
}
export async function getApiScanSN() {
  if (!apiScanSN)
    apiScanSN = await Swagger(process.env.REACT_APP_SCAN_SERIAL_NUMBERS)
  return apiScanSN
}
export async function getApiAuth(access_token) {
  return await Swagger(
    process.env.REACT_APP_SWAGGER_AUTH,
    requestOptions(access_token)
  )
}
export const storageSwaggerTag = 'Commissioning'
