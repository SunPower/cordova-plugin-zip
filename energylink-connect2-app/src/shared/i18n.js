import { useMemo } from 'react'
import { useSelector } from 'react-redux'

import * as languages from './locales'
import * as localizedComponents from './localizedComponents'

const getTranslate = translations => (key, ...params) => {
  let value = translations[key] || key
  if (params) {
    params.forEach((p, idx) => {
      value = value.replace(`{${idx}}`, p)
    })
  }
  return value
}

export const useI18n = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language
  return useMemo(() => getTranslate(languages[locale]), [locale])
}

export const useI18nComponent = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language || {}

  return useMemo(
    () => componentName => {
      const path = `${locale}_${componentName}`
      const Component = localizedComponents[path] || {}
      if (typeof Component !== 'function') {
        throw new Error(`Component not found: ${path}`)
      }
      return Component()
    },
    [locale]
  )
}
