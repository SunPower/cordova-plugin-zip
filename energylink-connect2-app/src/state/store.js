import { applyMiddleware, compose, createStore } from 'redux'
import { createEpicMiddleware } from 'redux-observable'
import { persistStore } from 'redux-persist'
import thunk from 'redux-thunk'
import rootReducer from './reducers'
import rootEpic from './epics'

const epicMiddleware = createEpicMiddleware()

export function configureStore(initialState) {
  const middlewares = []

  const composeEnhancers =
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

  middlewares.push(thunk)
  middlewares.push(epicMiddleware)

  if (
    process.env.NODE_ENV === 'development' ||
    process.env.REACT_APP_ENABLE_LOGGER
  ) {
    const { createLogger } = require('redux-logger')
    const logger = createLogger({
      predicate: (getState, action) => action.type !== 'DEVICE_IS_CONNECTED',
      collapsed: (getState, action, logEntry) => !logEntry.error
    })
    middlewares.push(logger)
  }

  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancers(applyMiddleware(...middlewares))
  )

  if (
    process.env.NODE_ENV === 'development' ||
    process.env.REACT_APP_ENABLE_LOGGER
  ) {
    window.store = store
  }

  epicMiddleware.run(rootEpic)

  const persistor = persistStore(store)
  return { store, persistor }
}
