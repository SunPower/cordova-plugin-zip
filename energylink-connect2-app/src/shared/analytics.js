import appVersion from '../macros/appVersion.macro'
import { capitalizeWord, getUserProfile } from 'shared/analyticsUtils'
import { MIXPANEL_EVENT_QUEUED } from 'state/actions/analytics'
import { getAppleDeviceFamily } from 'shared/appleDevicesTable'
import { prop } from 'ramda'

const getAppAndDeviceProperties = () => {
  const { device } = window
  const { model, version, platform, manufacturer } = device

  let deviceModel
  let deviceBrowserVersion

  if (platform === 'iOS') {
    deviceModel = getAppleDeviceFamily(model)
    deviceBrowserVersion = version
  } else {
    deviceModel = `${manufacturer}, ${model}`
  }

  return {
    'App Build': appVersion(),
    'App Flavor': process.env.REACT_APP_FLAVOR,
    'OS Version': version,
    ...(deviceBrowserVersion ? { $browser_version: deviceBrowserVersion } : {}),
    $device: deviceModel
  }
}

export const loggedIn = user => {
  const { mixpanel } = window
  const {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  } = getUserProfile(user)
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Login', { Success: true })

  return MIXPANEL_EVENT_QUEUED('Login - Success')
}

export const loginFailed = () => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.track('Login', { Success: false })

  return MIXPANEL_EVENT_QUEUED('Login - Failed')
}

export const scanPVS = ({ pvsSN, entryMethod }, event) => {
  const { mixpanel } = window

  mixpanel.register({ 'PVS SN': pvsSN })
  mixpanel.track('Scan PVS Tag', {
    'Entry Method': entryMethod
  })

  return MIXPANEL_EVENT_QUEUED(event)
}

export const siteNotFound = ({
  uniqueId: userId,
  firstName,
  lastName,
  email,
  dealerName,
  recordType: dealerType
}) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', { Found: false })

  return MIXPANEL_EVENT_QUEUED('Find Site - site not found')
}

export const siteFound = (
  {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  },
  { city, st_id, postalCode, siteKey, commissioned }
) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', {
    Found: true,
    ...{
      $city: city,
      State: st_id,
      'Zip Code': postalCode,
      'Site ID': siteKey,
      Commissioned: commissioned
    }
  })

  return MIXPANEL_EVENT_QUEUED('Find Site - site found')
}

export const saveConfiguration = config => {
  const { mixpanel } = window
  mixpanel.track('Configure', config)
  return MIXPANEL_EVENT_QUEUED('Configure - submit site config')
}

export const saveInventory = (inventory, hasEss) => {
  const { mixpanel } = window

  mixpanel.register({
    'Equinox Storage Site': hasEss
  })

  mixpanel.track('Inventory', {
    'AC Module Count': prop('AC_MODULES', inventory),
    'ESS System Size': hasEss ? prop('ESS', inventory) : 'None',
    'Equinox Storage Site': hasEss
  })
  return MIXPANEL_EVENT_QUEUED('Inventory')
}

export const deviceResume = pageResume => {
  const { mixpanel } = window
  mixpanel.register({
    Page: pageResume
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

/**
 *
 * @param {string}[interfaceType] - The type of connection the user has. either can be  PVS, Cell, or Customer WiFi
 * @return {Action<null, null>}
 */
export const internetConnection = (interfaceType = '') => {
  const { mixpanel } = window
  mixpanel.register({
    'Internet Connection': interfaceType
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

export const firmwareUpdateEvent = data => {
  const { mixpanel } = window
  mixpanel.track('Firmware Update', {
    'Current FW Version': prop('fromFWVersion', data),
    'Current BN Version': prop('fromBuildNumber', data),
    'New FW Version': prop('toFWVersion', data),
    'New BN Version': prop('toBuildNumber', data),
    Success: prop('success', data)
  })
  return MIXPANEL_EVENT_QUEUED('Firmware update - Update successful')
}

export const timeMixPanelEvent = name => {
  if (!name) throw new Error('The name parameter is required')
  const { mixpanel } = window
  mixpanel.time_event(name)
}

export const commissionSite = ({ duration, timeConfiguring }) => {
  const { mixpanel } = window
  mixpanel.track('Commission Site', {
    $duration: duration,
    'Time Configuring': timeConfiguring
  })
  return MIXPANEL_EVENT_QUEUED(
    'Commission Site - first successful site configuration'
  )
}
