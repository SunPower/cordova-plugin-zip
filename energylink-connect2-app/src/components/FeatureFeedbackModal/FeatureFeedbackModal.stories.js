import { storiesOf } from '@storybook/react'
import React from 'react'
import { Provider } from 'react-redux'

import FeatureFeedbackModal from '.'

import { configureStore } from 'state/store'

const featureFlagOn = {
  featureFlags: {
    featureFlags: [
      {
        lastUpdatedOn: '2020-10-01',
        name: 'ct-checks',
        page: 'feature-feedback',
        status: true
      }
    ]
  }
}

const featureFlagOff = {
  featureFlags: {
    featureFlags: [
      {
        lastUpdatedOn: '2020-10-01',
        name: 'ct-checks',
        page: 'feature-feedback',
        status: false
      }
    ]
  }
}

const featureFlagOnSending = {
  featureFlags: {
    featureFlags: [
      {
        lastUpdatedOn: '2020-10-01',
        name: 'ct-checks',
        page: 'feature-feedback',
        status: true
      }
    ]
  },
  global: {
    isSendingFeedback: true
  }
}

const featureFlagOnError = {
  featureFlags: {
    featureFlags: [
      {
        lastUpdatedOn: '2020-10-01',
        name: 'ct-checks',
        page: 'feature-feedback',
        status: true
      }
    ]
  },
  global: {
    feedbackError: true
  }
}

const featureFlagOnSent = {
  featureFlags: {
    featureFlags: [
      {
        lastUpdatedOn: '2020-10-01',
        name: 'ct-checks',
        page: 'feature-feedback',
        status: true
      }
    ]
  },
  global: {
    isFeedbackSuccessful: true
  }
}

storiesOf('FeatureFeedbackModal', module)
  .add('Open and feature flag on', () => {
    const { store } = configureStore(featureFlagOn)

    return (
      <Provider store={store}>
        <h1 className="mt-10 mb-10 has-text-centered is-uppercase">
          Feature feedback modal
        </h1>

        <div className="full-min-height pl-10 pr-10">
          <FeatureFeedbackModal
            featureFlagName="ct-checks"
            open={true}
            placeholder="This is the placeholder"
            title="This is the title"
          />
        </div>
      </Provider>
    )
  })
  .add('Closed and feature flag on', () => {
    const { store } = configureStore(featureFlagOn)

    return (
      <Provider store={store}>
        <h1 className="mt-10 mb-10 has-text-centered is-uppercase">
          Feature feedback modal
        </h1>

        <div className="full-min-height pl-10 pr-10">
          <FeatureFeedbackModal
            featureFlagName="ct-checks"
            open={false}
            placeholder="This is the placeholder"
            title="This is the title"
          />
        </div>
      </Provider>
    )
  })
  .add('Open and feature flag off', () => {
    const { store } = configureStore(featureFlagOff)

    return (
      <Provider store={store}>
        <h1 className="mt-10 mb-10 has-text-centered is-uppercase">
          Feature feedback modal
        </h1>

        <div className="full-min-height pl-10 pr-10">
          <FeatureFeedbackModal
            featureFlagName="ct-checks"
            open={true}
            placeholder="This is the placeholder"
            title="This is the title"
          />
        </div>
      </Provider>
    )
  })
  .add('Open and feature flag on, sending feedback', () => {
    const { store } = configureStore(featureFlagOnSending)

    return (
      <Provider store={store}>
        <h1 className="mt-10 mb-10 has-text-centered is-uppercase">
          Feature feedback modal
        </h1>

        <div className="full-min-height pl-10 pr-10">
          <FeatureFeedbackModal
            featureFlagName="ct-checks"
            open={true}
            placeholder="This is the placeholder"
            title="This is the title"
          />
        </div>
      </Provider>
    )
  })
  .add('Open and feature flag on, sending feedback error', () => {
    const { store } = configureStore(featureFlagOnError)

    return (
      <Provider store={store}>
        <h1 className="mt-10 mb-10 has-text-centered is-uppercase">
          Feature feedback modal
        </h1>

        <div className="full-min-height pl-10 pr-10">
          <FeatureFeedbackModal
            featureFlagName="ct-checks"
            open={true}
            placeholder="This is the placeholder"
            title="This is the title"
          />
        </div>
      </Provider>
    )
  })
  .add('Open and feature flag on, sending feedback success', () => {
    const { store } = configureStore(featureFlagOnSent)

    return (
      <Provider store={store}>
        <h1 className="mt-10 mb-10 has-text-centered is-uppercase">
          Feature feedback modal
        </h1>

        <div className="full-min-height pl-10 pr-10">
          <FeatureFeedbackModal
            featureFlagName="ct-checks"
            open={true}
            placeholder="This is the placeholder"
            title="This is the title"
          />
        </div>
      </Provider>
    )
  })
