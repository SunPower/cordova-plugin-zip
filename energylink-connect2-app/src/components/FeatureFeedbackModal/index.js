import clsx from 'clsx'
import { includes, pathOr } from 'ramda'
import React, { useEffect, useState } from 'react'
import { useField, useForm } from 'react-final-form-hooks'
import { useDispatch, useSelector } from 'react-redux'

import Rating from 'components/Rating'
import TextArea from 'components/TextArea'
import SwipeableSheet from 'hocs/SwipeableSheet'
import { useFeatureFlag } from 'shared/featureFlags'
import { useI18n } from 'shared/i18n'
import { cleanString, either } from 'shared/utils'
import {
  FEATURE_FEEDBACK_MODAL_OPEN,
  SEND_FEEDBACK_INIT
} from 'state/actions/feedback'

const FeatureFeedbackModal = ({
  featureFlagName,
  open,
  placeholder,
  title
}) => {
  const t = useI18n()
  const dispatch = useDispatch()

  const zIndex = { zIndex: '9999' }

  const FEATURE_FLAG_PAGE = 'feature-feedback'
  const showFeedbackForm = useFeatureFlag({
    page: FEATURE_FLAG_PAGE,
    name: featureFlagName
  })
  const [showFeedbackModal, setShowFeedbackModal] = useState(false)

  const [feedbackSent, setFeedbackSent] = useState(false)

  const onSubmit = values => {
    const { comment } = values
    const sanitizedValues = { ...values, comment: cleanString(comment) }

    dispatch(SEND_FEEDBACK_INIT(sanitizedValues))
  }
  const changeRate = value => form.change('rating', value)

  const { form, handleSubmit } = useForm({
    onSubmit,
    initialValues: {
      comment: '',
      feature: true,
      rating: 1,
      source: `${FEATURE_FLAG_PAGE}/${featureFlagName}`
    }
  })
  const ratingField = useField('rating', form)
  const commentField = useField('comment', form)
  const { value: ratingValue } = ratingField.input
  const { input: commentInput, meta: commentMeta } = commentField

  const isSendingFeedback = useSelector(
    pathOr(false, ['global', 'isSendingFeedback'])
  )
  const feedbackSuccessfullySent = useSelector(
    pathOr(false, ['global', 'isFeedbackSuccessful'])
  )
  const sendingFeedbackError = useSelector(
    pathOr(false, ['global', 'feedbackError'])
  )
  const alreadyShownModals = useSelector(
    pathOr([], ['featureFeedback', 'alreadyShownModals'])
  )

  const submitClassnames = clsx('button', 'is-uppercase', 'is-primary', {
    'is-loading': isSendingFeedback
  })

  if (!feedbackSent && feedbackSuccessfullySent) {
    setFeedbackSent(true)
  }

  useEffect(() => {
    if (
      open &&
      showFeedbackForm &&
      !includes(featureFlagName, alreadyShownModals)
    ) {
      setShowFeedbackModal(true)
      dispatch(FEATURE_FEEDBACK_MODAL_OPEN({ featureFlagName }))
    }
  }, [alreadyShownModals, dispatch, featureFlagName, open, showFeedbackForm])

  return (
    <SwipeableSheet
      open={showFeedbackModal && !feedbackSent}
      onChange={() => setShowFeedbackModal(!showFeedbackModal)}
      style={zIndex}
    >
      <div className="feedback-modal has-text-centered mt-10 mb-20">
        <form onSubmit={handleSubmit} className="mt-20 pt-20">
          <div className="has-text-centered has-text-white has-text-weight-bold mb-5">
            {title}
          </div>

          <Rating onClick={changeRate} rating={ratingValue} />

          <TextArea
            input={commentInput}
            meta={commentMeta}
            placeholder={placeholder}
          />

          <div className="is-flex file level space-around pt-0 mt-20">
            <button className={submitClassnames} type="submit">
              {t('SUBMIT')}
            </button>
          </div>

          {either(
            sendingFeedbackError,
            <div className="message error mb-10 mt-10">
              <p className="pl-20 pr-20">{t('FEEDBACK_ERROR')}</p>
            </div>,
            null
          )}
        </form>
      </div>
    </SwipeableSheet>
  )
}

export default FeatureFeedbackModal
