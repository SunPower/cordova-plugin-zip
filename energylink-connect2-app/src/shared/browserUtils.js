import { createExternalLinkHandler } from 'shared/routing'

export function openUrl(url) {
  window.SafariViewController.isAvailable(function(available) {
    if (available) {
      window.SafariViewController.show({
        url: url,
        hidden: false, // default false. You can use this to load cookies etc in the background (see issue #1 for details).
        animated: false, // default true, note that 'hide' will reuse this preference (the 'Done' button will always animate though)
        transition: 'curl', // (this only works in iOS 9.1/9.2 and lower) unless animated is false you can choose from: curl, flip, fade, slide (default)
        enterReaderModeIfAvailable: false, // default false
        tintColor: '#ffffff', // default is ios blue
        barColor: '#15202e', // on iOS 10+ you can change the background color as well
        controlTintColor: '#15202e' // on iOS 10+ you can override the default tintColor
      })
    } else {
      // potentially powered by InAppBrowser because that (currently) clobbers window.open
      createExternalLinkHandler(url)()
    }
  })
}

export function dismissBrowser() {
  window.SafariViewController.hide()
}
