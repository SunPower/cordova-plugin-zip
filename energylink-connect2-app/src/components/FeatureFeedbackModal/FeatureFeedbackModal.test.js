import { shallow } from 'enzyme'
import React from 'react'
import * as reactRedux from 'react-redux'

import FeatureFeedbackModal from '.'

import * as i18n from 'shared/i18n'
import { configureStore } from 'state/store'

const featureFlagOn = {
  featureFlags: {
    featureFlags: [
      {
        lastUpdatedOn: '2020-10-01',
        name: 'ct-checks',
        page: 'feature-feedback',
        status: true
      }
    ]
  }
}

const featureFlagOff = {
  featureFlags: {
    featureFlags: [
      {
        lastUpdatedOn: '2020-10-01',
        name: 'ct-checks',
        page: 'feature-feedback',
        status: false
      }
    ]
  }
}

const featureFlagOnSending = {
  featureFlags: {
    featureFlags: [
      {
        lastUpdatedOn: '2020-10-01',
        name: 'ct-checks',
        page: 'feature-feedback',
        status: true
      }
    ]
  },
  global: {
    isSendingFeedback: true
  }
}

const featureFlagOnError = {
  featureFlags: {
    featureFlags: [
      {
        lastUpdatedOn: '2020-10-01',
        name: 'ct-checks',
        page: 'feature-feedback',
        status: true
      }
    ]
  },
  global: {
    feedbackError: true
  }
}

const featureFlagOnSent = {
  featureFlags: {
    featureFlags: [
      {
        lastUpdatedOn: '2020-10-01',
        name: 'ct-checks',
        page: 'feature-feedback',
        status: true
      }
    ]
  },
  global: {
    isFeedbackSuccessful: true
  }
}

describe('FeatureFeedbackModal component', () => {
  let dispatchMock

  beforeEach(() => {
    dispatchMock = jest.fn()
    jest.spyOn(reactRedux, 'useSelector').mockImplementation(() => dispatchMock)
    jest
      .spyOn(i18n, 'useI18n')
      .mockImplementation(path => (key, ...params) =>
        `${key.toUpperCase()} ${params.join('_')}`.trim()
      )
  })

  test('renders correctly when modal is open and feature flag is on', () => {
    const { store } = configureStore(featureFlagOn)
    const component = shallow(
      <reactRedux.Provider store={store}>
        <FeatureFeedbackModal
          featureFlagName="ct-checks"
          open={true}
          placeholder="This is the placeholder"
          title="This is the title"
        />
      </reactRedux.Provider>
    )
    expect(component).toMatchSnapshot()
  })

  test('renders correctly when modal is closed and feature flag is on', () => {
    const { store } = configureStore(featureFlagOn)
    const component = shallow(
      <reactRedux.Provider store={store}>
        <FeatureFeedbackModal
          featureFlagName="ct-checks"
          open={false}
          placeholder="This is the placeholder"
          title="This is the title"
        />
      </reactRedux.Provider>
    )
    expect(component).toMatchSnapshot()
  })

  test('renders correctly when modal is open and feature flag is off', () => {
    const { store } = configureStore(featureFlagOff)
    const component = shallow(
      <reactRedux.Provider store={store}>
        <FeatureFeedbackModal
          featureFlagName="ct-checks"
          open={true}
          placeholder="This is the placeholder"
          title="This is the title"
        />
      </reactRedux.Provider>
    )
    expect(component).toMatchSnapshot()
  })

  test('renders correctly when modal is open, feature flag is on and feedback is sending', () => {
    const { store } = configureStore(featureFlagOnSending)
    const component = shallow(
      <reactRedux.Provider store={store}>
        <FeatureFeedbackModal
          featureFlagName="ct-checks"
          open={true}
          placeholder="This is the placeholder"
          title="This is the title"
        />
      </reactRedux.Provider>
    )
    expect(component).toMatchSnapshot()
  })

  test('renders correctly when modal is open, feature flag is on and feedback error occurs', () => {
    const { store } = configureStore(featureFlagOnError)
    const component = shallow(
      <reactRedux.Provider store={store}>
        <FeatureFeedbackModal
          featureFlagName="ct-checks"
          open={true}
          placeholder="This is the placeholder"
          title="This is the title"
        />
      </reactRedux.Provider>
    )
    expect(component).toMatchSnapshot()
  })

  test('renders correctly when modal is open, feature flag is on and feedback is successfully sent', () => {
    const { store } = configureStore(featureFlagOnSent)
    const component = shallow(
      <reactRedux.Provider store={store}>
        <FeatureFeedbackModal
          featureFlagName="ct-checks"
          open={true}
          placeholder="This is the placeholder"
          title="This is the title"
        />
      </reactRedux.Provider>
    )
    expect(component).toMatchSnapshot()
  })
})
