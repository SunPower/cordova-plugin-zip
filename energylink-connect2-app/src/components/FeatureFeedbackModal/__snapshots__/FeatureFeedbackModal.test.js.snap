// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureFeedbackModal component renders correctly when modal is closed and feature flag is on 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <FeatureFeedbackModal
    featureFlagName="ct-checks"
    open={false}
    placeholder="This is the placeholder"
    title="This is the title"
  />
</ContextProvider>
`;

exports[`FeatureFeedbackModal component renders correctly when modal is open and feature flag is off 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <FeatureFeedbackModal
    featureFlagName="ct-checks"
    open={true}
    placeholder="This is the placeholder"
    title="This is the title"
  />
</ContextProvider>
`;

exports[`FeatureFeedbackModal component renders correctly when modal is open and feature flag is on 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <FeatureFeedbackModal
    featureFlagName="ct-checks"
    open={true}
    placeholder="This is the placeholder"
    title="This is the title"
  />
</ContextProvider>
`;

exports[`FeatureFeedbackModal component renders correctly when modal is open, feature flag is on and feedback error occurs 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <FeatureFeedbackModal
    featureFlagName="ct-checks"
    open={true}
    placeholder="This is the placeholder"
    title="This is the title"
  />
</ContextProvider>
`;

exports[`FeatureFeedbackModal component renders correctly when modal is open, feature flag is on and feedback is sending 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <FeatureFeedbackModal
    featureFlagName="ct-checks"
    open={true}
    placeholder="This is the placeholder"
    title="This is the title"
  />
</ContextProvider>
`;

exports[`FeatureFeedbackModal component renders correctly when modal is open, feature flag is on and feedback is successfully sent 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <FeatureFeedbackModal
    featureFlagName="ct-checks"
    open={true}
    placeholder="This is the placeholder"
    title="This is the title"
  />
</ContextProvider>
`;
